runtime/permissions.rs

```diff
@@ -1319,4 +1351,61 @@ mod tests {
       assert_eq!(perms.hrtime.revoke(), PermissionState::Denied);
     };
   }
+
+  #[test]
+  fn test_check() {
+    let mut perms = Permissions {
+      read: Permissions::new_read(&None, true),
+      write: Permissions::new_write(&None, true),
+      net: Permissions::new_net(&None, true),
+      env: Permissions::new_env(false, true),
+      run: Permissions::new_run(&None, true),
+      plugin: Permissions::new_plugin(false, true),
+      hrtime: Permissions::new_hrtime(false, true),
+    };
+
+    let _guard = PERMISSION_PROMPT_GUARD.lock().unwrap();
+
+    set_prompt_result(false);
+    assert!(perms.read.check(&Path::new("/foo")).is_err());
+    set_prompt_result(true);
+    assert!(perms.read.check(&Path::new("/foo")).is_ok());
```

Comment:
This behaviour is incorrect. After the first check fails it should be put into the denied state, the second should bypass the prompt and return an error as well.